packages/hub/routes/job-tickets.ts

```diff
@@ -131,18 +119,20 @@ export default class JobTicketsRoute {
       return;
     }
 
-    this.workerClient.addJob(jobTicketToRetry.jobType, jobTicketToRetry.payload, jobTicketToRetry.spec);
-
-    let newJobTicket: JobTicket | null = {
-      id: shortUUID.uuid(),
-      jobType: jobTicketToRetry.jobType,
-      ownerAddress: jobTicketToRetry.ownerAddress,
-      payload: jobTicketToRetry.payload,
-      result: null,
-      spec: jobTicketToRetry.spec,
-    };
-
-    newJobTicket = await this.jobTicketsQueries.insert(newJobTicket);
+    let taskName = jobTicketToRetry.job_type as unknown as keyof KnownTasks;
```

Comment:
Is there a way to type this more strongly?

Reply:
I can’t think of a good way, it would have to be guaranteed that when we insert a ticket the type is a known task type 🤔

Reply:
runtime-only though, right? Better than nothing!

Reply:
it's a runtime check, and also removes the necessity of doing type assertions (`as Type`):

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates